packages/apollo-cli/src/commands/schema/checkQueries.ts

```diff
@@ -0,0 +1,146 @@
+  };
+
+  async run() {
+    const { flags } = this.parse(CheckQueries);
+
+    const apiKey = process.env.ENGINE_API_KEY || flags.key;
+    const pullFromEngine = !!apiKey;
+
+    const tasks = new Listr([
+      {
+        title: "Scanning for GraphQL queries",
+        task: async (ctx, task) => {
+          const paths = await globby(flags.queries ? [ flags.queries ] : []);
+          task.title = `Scanning for GraphQL queries (${paths.length} found)`;
+          ctx.queryPaths = paths;
+        }
+      },
+      {
+        title: pullFromEngine ? "Loading schema from Apollo Engine" : "Loading GraphQL schema",
+        task: async ctx => {
+          if (pullFromEngine) {
+            const variables = {
+              id: getIdFromKey(apiKey as string),
+              tag: flags.tag || "current",
+            }
+
+            const engineSchema = await toPromise(
+              execute(engineLink, {
+                query: SCHEMA_QUERY,
+                variables,
+                context: {
+                  headers: { ["x-api-key"]: apiKey },
+                  ...(flags.engine && { uri: flags.engine }),
+                },
+              })
+            );
+
+            if (engineSchema.data && engineSchema.data.service.schema) {
+              ctx.schema = buildClientSchema(
+                engineSchema.data.service.schema
+              );
+            } else {
+              this.error("Failed to get schema from Apollo Engine");
+            }
+          } else {
+            const schemaFileContent = await promisify(fs.readFile)(path.resolve(flags.schema as string));
+            const schemaData = JSON.parse(schemaFileContent as any as string);
+            ctx.schema = buildClientSchema(
+              (schemaData.data) ? schemaData.data : (
+                schemaData.__schema ? schemaData : { __schema: schemaData }
+              )
+            )
+          }
+        }
+      },
+      {
+        title: "Checking query compatibility with schema",
+        task: async (ctx, task) => {
+          const docs = loadQueryDocuments(ctx.queryPaths, flags.tagName);
+
+          const errors: GraphQLError[] = [];
+          const warnings: GraphQLError[] = []
+
+          docs.forEach((doc, i) => {
+            const validateResult = validate(ctx.schema, doc);
+            const deprecationResult = findDeprecatedUsages(ctx.schema, doc);
+
+            errors.push(...validateResult);
+            warnings.push(...deprecationResult);
+
+            task.title = `Checking query compatibility with schema (${i + 1}/${docs.length}, ${errors.length} errors, ${warnings.length} warnings)`;
+            return validateResult.concat(deprecationResult);
+          });
+
+          if (errors.length > 0 || warnings.length > 0) {
+            throw { errors, warnings };
+          }
+        },
+      },
+    ]);
+
+    return tasks.run().catch(({ errors, warnings }) => {
+      if (warnings.length > 0) {
+        this.warn(warnings.join("\n"));
```

Comment:
lets give these both some padding around the title (i.e. a new line before and after?

can we give the full file name?